lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -104,29 +218,37 @@ void LowerLayersPass::runOnModule(FModuleOp moduleOp) {
     // block.
     SmallVector<PortInfo> ports;
 
-    // Connectsion that need to be made to the instance of the derived module.
-    SmallVector<Value> connectValues;
+    // Connection that need to be made to the instance of the derived module.
+    SmallVector<ConnectInfo> connectValues;
 
     // Create an input port for an operand that is captured from outside.
-    //
-    // TODO: If we allow capturing reference types, this will need to be
-    // updated.
     auto createInputPort = [&](Value operand, Location loc) {
       auto portNum = ports.size();
       auto operandName = getFieldName(FieldRef(operand, 0), true);
 
-      ports.push_back({builder.getStringAttr("_" + operandName.first),
-                       operand.getType(), Direction::In, /*sym=*/{},
+      // The type be a non-ref.
+      auto type = operand.getType();
```

Comment:
`auto type = getBaseType(operand.getType())` ?

Although that's nullable now so maybe nevermind.